app/src/main/java/com/example/admin/learningandroid/threads/ApiCallUsingRetrofit.java

```diff
@@ -0,0 +1,142 @@
+package com.example.admin.learningandroid.threads;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.DividerItemDecoration;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.example.admin.learningandroid.R;
+import com.example.admin.learningandroid.entity.ListDataRetrieveModelClass;
+import com.example.admin.learningandroid.entity.RetrofitWeatherApiModelClass;
+import com.example.admin.learningandroid.entity.Weather;
+import com.example.admin.learningandroid.entity.WeatherAPIModelClass;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+import static com.example.admin.learningandroid.Constants.BASE_WEATHER_URL;
+
+public class ApiCallUsingRetrofit extends AppCompatActivity {
+    ArrayList<WeatherAPIModelClass> weatherAPIModelClassArrayList = new ArrayList<>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.retrofit_weather_api_layout);
+        final RecyclerView weatherDataDisplay = findViewById(R.id.retrofit_weather_api_recycler_view);
+
+        Retrofit dataRetreive = new Retrofit.Builder()
+                .baseUrl(ApiCall.weatherURL)
+                .addConverterFactory(GsonConverterFactory
+                        .create()).build();
+        ApiCall apiCall = dataRetreive.create(ApiCall.class);
+        Call<ListDataRetrieveModelClass> listDataRetrieveModelClassCall = apiCall.getWeatherData();
```

Comment:
ListDataRetrieveModelClass is a  class its self explanatory please avoid appending class prefix